File: haskell/quicksort.hs
import Data.List

quicksort :: [Int] -> [Int]
quicksort [] = []
quicksort (pivot : xs) =
  let (lower, upper) = partition (< pivot) xs
  in quicksort lower ++ [pivot] ++ quicksort upper

main = do n <- getLine
          arr <- getLine
          let sorted = quicksort (map read (words arr))
          putStrLn (unwords (map show sorted))

File: erlang/quicksort.erl
-module(quicksort).

quicksort([]) -> []; 
quicksort([Pivot|Rest]) ->
  {Lower,Upper} = lists:partition(fun(A) -> A < Pivot end, Rest),
  quicksort(Lower) ++ [Pivot] ++ quicksort(Upper).

read_int() -> {_,[N]} = io:fread("", "~d"), N.

main(_) ->
  N = read_int(),
  List = [read_int() || _ <- lists:seq(1, N)],
  Sorted = quicksort(List),
  io:fwrite("~s~n", [string:join([integer_to_list(X) || X <- Sorted], " ")]).
